editor/modes/visual-editor/block.js

```diff
@@ -270,7 +313,81 @@ class VisualEditorBlock extends Component {
 
 	onKeyUp( event ) {
 		this.removeOrDeselect( event );
-		this.handleArrowKey( event );
+		//this.handleArrowKey( event );
+	}
+
+	handleToolbarTabCycle( event ) {
+		const { keyCode, shiftKey, target } = event;
```

Comment:
Maybe we can create a separate component which manages groups of toolbars and the keyboard navigation between them? This code doesn't seem particularly relevant to block rendering.

That component could then listen to changes in block `focus` state.